colosseum/declaration.py

```diff
@@ -1,29 +1,87 @@
     BORDER_SPACING_CHOICES, BORDER_STYLE_CHOICES, BORDER_WIDTH_CHOICES,
     BOX_OFFSET_CHOICES, CAPTION_SIDE_CHOICES, CLEAR_CHOICES, CLIP_CHOICES,
-    COLOR_CHOICES, DIRECTION_CHOICES, DISPLAY_CHOICES, EMPTY_CELLS_CHOICES,
-    FLEX_BASIS_CHOICES, FLEX_DIRECTION_CHOICES, FLEX_GROW_CHOICES,
-    FLEX_SHRINK_CHOICES, FLEX_START, FLEX_WRAP_CHOICES, FLOAT_CHOICES,
-    GRID_AUTO_CHOICES, GRID_AUTO_FLOW_CHOICES, GRID_GAP_CHOICES,
-    GRID_PLACEMENT_CHOICES, GRID_TEMPLATE_AREA_CHOICES, GRID_TEMPLATE_CHOICES,
-    INLINE, JUSTIFY_CONTENT_CHOICES, LETTER_SPACING_CHOICES, LTR,
-    MARGIN_CHOICES, MAX_SIZE_CHOICES, MIN_SIZE_CHOICES, NORMAL, NOWRAP,
-    ORDER_CHOICES, ORPHANS_CHOICES, OVERFLOW_CHOICES, PADDING_CHOICES,
+    COLOR_CHOICES, DIRECTION_CHOICES, DISPLAY_CHOICES,
+    EMPTY_CELLS_CHOICES, FLEX_BASIS_CHOICES, FLEX_DIRECTION_CHOICES,
+    FLEX_GROW_CHOICES, FLEX_SHRINK_CHOICES, FLEX_START, FLEX_WRAP_CHOICES,
+    FLOAT_CHOICES, GRID_AUTO_CHOICES, GRID_AUTO_FLOW_CHOICES,
+    GRID_GAP_CHOICES, GRID_PLACEMENT_CHOICES, GRID_TEMPLATE_AREA_CHOICES,
+    GRID_TEMPLATE_CHOICES, INITIAL_OUTLINE_VALUES, INLINE, INVERT,
+    JUSTIFY_CONTENT_CHOICES, LETTER_SPACING_CHOICES, LTR, MARGIN_CHOICES,
+    MAX_SIZE_CHOICES, MEDIUM, MIN_SIZE_CHOICES, NORMAL, NOWRAP, ORDER_CHOICES,
+    ORPHANS_CHOICES, OUTLINE_COLOR_CHOICES, OUTLINE_STYLE_CHOICES,
+    OUTLINE_WIDTH_CHOICES, OVERFLOW_CHOICES, PADDING_CHOICES,
     PAGE_BREAK_AFTER_CHOICES, PAGE_BREAK_BEFORE_CHOICES,
     PAGE_BREAK_INSIDE_CHOICES, POSITION_CHOICES, ROW, SEPARATE, SHOW,
     SIZE_CHOICES, STATIC, STRETCH, TABLE_LAYOUT_CHOICES, TEXT_ALIGN_CHOICES,
     TEXT_DECORATION_CHOICES, TEXT_INDENT_CHOICES, TEXT_TRANSFORM_CHOICES, TOP,
     TRANSPARENT, UNICODE_BIDI_CHOICES, VISIBILITY_CHOICES, VISIBLE,
-    WHITE_SPACE_CHOICES, WIDOWS_CHOICES, WORD_SPACING_CHOICES, Z_INDEX_CHOICES,
-    OtherProperty, TextAlignInitialValue, default,
+    WHITE_SPACE_CHOICES, WIDOWS_CHOICES, WORD_SPACING_CHOICES,
+    Z_INDEX_CHOICES, OtherProperty, TextAlignInitialValue, default,
 )
+from .exceptions import ValidationError
+from .wrappers import Outline
 
 _CSS_PROPERTIES = set()
 
 
+def validated_shorthand_property(name, initial, parser, storage_class):
+    """Define the shorthand CSS font property."""
+    if not isinstance(initial, dict):
+        raise ValueError('Initial values for shorthand property must be of type dict!')
+
+    initial = storage_class(**initial)
+
+    def getter(self):
+        shorthand = storage_class()
+        for property_name in initial:
+            try:
+                property_value = getattr(self, '_%s' % property_name)
+                shorthand[property_name] = property_value
+            except AttributeError:
+                pass
+
+        if shorthand:
+            return shorthand
+        else:
+            # It no property has been defined for the outline, return initial values
+            return initial.copy()
```

Comment:
corner case, What if no propeties have been defined at all? I figured returning the full initial values made sense

Reply:
If we decide that this should return the initial values we could actually query them an there would be no need to use the initial values dict. In that case we still need to know that keys to iterate on.

To avoid duplicating the definition of the initial values inside a dict and inside the actual properties then we need to pass the properties, in this case 

the call to validated_shorthand_proerty copuld something like

```python
	def validated_shorthand_property(name, properties, parser, storage_class)
```

Or we oculd define the keys as part of the strorage class (which we actually already do) so it could be as simple as

```python
	def validated_shorthand_property(name, parser, storage_class)
```

This is actually my favorite option since we do not define initial properties on 2 separate spots and we keep the function call simpler.

@freakboy3742 If you agree I can refactor the code a bit so that it simplifies to this option.